Simplify and update logic related to serialization

pjanik opened the pull request:
This PR makes several adjustments to the serialization logic.

The biggest change is reducing the number of handled events. We don’t need to worry about `fireEventAdded` or `sparkAdded` anymore since the snapshot is taken at the end of the fire event. @eireland, you were probably seeing zero sparks because you were emitting this event after the sparks array was cleared. I’ve moved the event emitter line up a few lines to fix this.

The timeline now snaps to snapshots only instead of always jumping one year. This change helps with orientation (you won't experience moving the handle without any effect) and makes finding fire event snapshots easier.

I’ve also removed some `Math.floor` that were scattered around when dealing with year value. This ensures the handle's position aligns better with the fire event position.

Additionally, there are changes related to how snapshots are saved and restored. Some logic has been moved out of the Timeline's `useEffect` to the snapshot manager.

Lastly, I added a 2 year margin between fire event snapshots and regular snapshots, making them easier to locate.

eireland reviewed (approved):
👍 LGTM. Except when fire event is within year 1 (start the sim with a fire event), and you scrub back to it, it shows the sparks without the burn areas? Not sure what the team want to show though.
<img width="1237" alt="image" src="https://github.com/user-attachments/assets/a0ce9692-c9d5-4ca9-b5e0-386bb1914809">

And average temp condition during fire event is not restored on scrub back. Should average temp go back to default while sim is running or should it stay in the same temp?

pjanik commented:
@eireland, thank you. That's a good test scenario for starting the simulation with a fire event. It's now fixed. I also addressed the issue with the average temperature.

In the last commit, I fixed an issue with the timeline time progress. I managed to remove the `timeProgress` state, and the `useEffect` was reduced to a single line.
